src/arch/host/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ target_include_directories(sof_public_headers INTERFACE
 endif()
 
 # C & ASM flags
-target_compile_options(sof_options INTERFACE -g -O3 -Wall -Werror -Wmissing-prototypes
```

Comment:
what's the default one? -O1?

Reply:
I hope we could get to cmake -DCMAKE_BUILD_TYPE=Release and Debug to control this. Release could be default with -O2 or -O3 and Debug without optimization to get linear code stepping in gdb/xt-gdb and no <optimized out> variables.

Reply:
I would suggest
1. keep default as same with sof.
2. add option for optimization level for debug usage.

This is due to we also want to add performance counter, and this performance counter may not be same with real time, 
if we set to o1, then fw overhead cost will be more than realtime, this will cause us more confusion from performance perspective.

Reply:
We should stop setting `-g` and `-O` flags manually. There is `-DCMAKE_BUILD_TYPE` for that and Zephyr has a (different) toolchain framework too.

The problem is legacy XTOS

Reply:
I've removed changing the build options from this PR, it's not relevant for this work. I hope the build for debugging vs. evaluating optimized performance can be solved with other work and PR.